Core/Systems/DummyTileSystem/DummySystem.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace StarlightRiver.Core.Systems.DummyTileSystem
+{
+	internal class DummySystem : ModSystem
+	{
+		public static List<Dummy> dummies = new();
+		public static Dictionary<int, Dummy> prototypes = new();
+		public static Dictionary<Type, int> types = new();
+
+		public override void Load()
+		{
+			On_Main.DrawProjectiles += DrawDummies;
+			On_Main.DoDraw_DrawNPCsBehindTiles += DrawBehindNPCs;
+		}
+
+		public override void PostUpdateProjectiles()
+		{
+			dummies.ForEach(n => n.AI());
+
+			dummies.RemoveAll(n => !n.active);
+		}
+
+		private void DrawDummies(On_Main.orig_DrawProjectiles orig, Main self)
+		{
+			orig(self);
+
+			Main.spriteBatch.Begin(SpriteSortMode.Deferred, BlendState.AlphaBlend, SamplerState.PointClamp, DepthStencilState.Default, RasterizerState.CullNone, null, Main.Transform);
+
+			dummies.ForEach(n =>
+			{
+				if (!n.offscreen)
+					n.PostDraw(Lighting.GetColor((n.Center / 16).ToPoint()));
+			});
+
+			Main.spriteBatch.End();
+		}
+
+		private void DrawBehindNPCs(On_Main.orig_DoDraw_DrawNPCsBehindTiles orig, Main self)
+		{
+			Main.spriteBatch.Begin(SpriteSortMode.Deferred, BlendState.AlphaBlend, SamplerState.PointClamp, DepthStencilState.Default, RasterizerState.CullNone, null, Main.Transform);
+
+			dummies.ForEach(n =>
+			{
+				if (!n.offscreen)
+					n.DrawBehindTiles();
+			});
+
+			Main.spriteBatch.End();
+
+			orig(self);
+		}
+
+		public override void Unload()
+		{
+			dummies.Clear();
+			prototypes.Clear();
+			types.Clear();
+		}
+
+		public override void OnWorldLoad()
+		{
+			dummies.Clear();
+		}
+
+		/// <summary>
+		/// Spawns a new dummy at the given point in the world
+		/// </summary>
+		/// <param name="type">The numeric type of the dummy to spawn</param>
+		/// <param name="pos">The position in the world to spawn the dummy at</param>
+		public static Dummy NewDummy(int type, Vector2 pos)
+		{
+			Dummy dummy = prototypes[type].Clone();
+			dummy.active = true;
+			dummy.Center = pos;
+			dummy.identity = Main.rand.Next(int.MaxValue); // Lets hope this is random enough!
```

Comment:
this is not guaranteed to be unique. just get like the current system time or something of that nature or implement a autonumber system if that's cyclic